sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -480,7 +480,16 @@ case class FileSourceScanExec(
           partitionValues = partition.values
         )
       }
-    }.sortBy(_.length)(implicitly[Ordering[Long]].reverse)
+    }
+
+    val orderByFilePath = fsRelation.sparkSession.sessionState.conf.orderByFilePathWhenPartitioning
```

Comment:
`conf.orderByFilePathWhenPartitioning`?

Reply:
@maropu Do you mean this name is not suitable?